java/clients/src/main/java/sleeper/status/report/CompactionJobStatusReport.java

```diff
@@ -0,0 +1,258 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sleeper.status.report;
+
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB;
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClientBuilder;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import sleeper.ClientUtils;
+import sleeper.compaction.job.CompactionJobStatusStore;
+import sleeper.compaction.job.status.CompactionJobStatus;
+import sleeper.compaction.status.job.DynamoDBCompactionJobStatusStore;
+import sleeper.configuration.properties.InstanceProperties;
+import sleeper.status.report.compactionjob.CompactionJobStatusCollector;
+import sleeper.status.report.compactionjob.CompactionJobStatusReporter;
+import sleeper.status.report.compactionjob.CompactionJobStatusReporter.QueryType;
+import sleeper.status.report.compactionjob.JsonCompactionJobStatusReporter;
+import sleeper.status.report.compactionjob.StandardCompactionJobStatusReporter;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Scanner;
+
+import static sleeper.configuration.properties.UserDefinedInstanceProperty.ID;
+
+public class CompactionJobStatusReport {
+    private final CompactionJobStatusReporter compactionJobStatusReporter;
+    private final CompactionJobStatusCollector compactionJobStatusCollector;
+    private static final String DEFAULT_STATUS_REPORTER = "STANDARD";
+    private static final Map<String, CompactionJobStatusReporter> FILE_STATUS_REPORTERS = new HashMap<>();
+    private static final Instant DEFAULT_RANGE_START = Instant.now().minus(4L, ChronoUnit.HOURS);
+    private static final Instant DEFAULT_RANGE_END = Instant.now();
+
+    static {
+        FILE_STATUS_REPORTERS.put(DEFAULT_STATUS_REPORTER, new StandardCompactionJobStatusReporter());
+        FILE_STATUS_REPORTERS.put("JSON", new JsonCompactionJobStatusReporter());
+    }
+
+    private static final Map<String, QueryType> QUERY_TYPES = new HashMap<>();
+
+    static {
+        QUERY_TYPES.put("-d", QueryType.DETAILED);
+        QUERY_TYPES.put("-r", QueryType.RANGE);
+        QUERY_TYPES.put("-u", QueryType.UNFINISHED);
```

Comment:
Could we have a query type of all to just dump all information about all compaction jobs?